src/lsm/manifest_level.zig

```diff
@@ -256,29 +258,22 @@ pub fn ManifestLevelType(
 
             table.snapshot_max = snapshot;
             level.table_count_visible -= 1;
-            level.key_range.exclude(KeyRange{
+            level.key_range_latest.exclude(KeyRange{
                 .key_min = table.key_min,
                 .key_max = table.key_max,
             });
         }
 
         /// Remove the given table.
-        /// - If the exact table does not exist in the ManifestLevel: return null.
-        /// - If the exact table exists in the ManifestLevel: remove it, and return whether the
-        ///   table is visible to any of the given snapshots or `snapshot_latest`.
+        /// Returns whether the exact table was present, (and therefore removed).
```

Comment:
Maybe just assert that the table is present? I don't think removing non-existant table would make sense, as there are just no tables outside the manifest (and we key removal by the table itself, not by a key)